### DISCUSSION TITLE
UnhandledRejection Non-Error promise rejection captured with value: Object Not Found Matching Id:

### DISCUSSION THREAD

**Discussion by Alvaro-c** [getsentry/support#28] [Support] [answered]
Hi everyone, 

First of all, sorry if there is already an **open** discussion going on with this issue. 

I opening this new thread because the last one I found was closed: 
https://forum.sentry.io/t/unhandledrejection-non-error-promise-rejection-captured-with-value/14062 (closed)
And in that very thread GitHub Discussions is mentioned (take this as a disclaimer if this is not the right place to open this): 
![image](https://user-images.githubusercontent.com/78445291/176183683-b2c48aa9-5258-4a8d-97e8-8fb55293bd67.png)
(If I should open an issue, please let me know)

So far, the info I collected is the following:

Sentry is reporting the error described in the title. 

Issue could be caused by an Outlook bot checking links, according to the user [abhishek1234321](https://github.com/abhishek1234321), seen in the thread https://github.com/getsentry/sentry-javascript/issues/3440 (closed)

Issue is not addressed and according to this post, Sentry suggest to filter out the issue with the following code: 

![image](https://user-images.githubusercontent.com/78445291/176182057-a123e487-71c3-435b-bbf4-dccd1aa62258.png)

`Sentry.init({
    dsn:__MY_DSN__,
      ignoreErrors:[
       "Non-Error exception captured",
       "Non-Error promise rejection captured"
      ]
    })`

Seen here https://forum.sentry.io/t/unhandledrejection-non-error-promise-rejection-captured-with-value/14062/16

According to the user [Blackth0rn](https://forum.sentry.io/u/Blackth0rn), maybe the issue is coming from this code: 
[cefsharp/CefSharp/blob/445fb9ab2d178b55e3ca6561722d151ce3d9337e/CefSharp/Internals/JavascriptObjectRepository.cs#L293 100](https://github.com/cefsharp/CefSharp/blob/445fb9ab2d178b55e3ca6561722d151ce3d9337e/CefSharp/Internals/JavascriptObjectRepository.cs#L293)

Apart from that, issue is not addressed and I am not finding much more to solve it. 

Any help is appreciated :) 

EDIT: 
Additional findings: 
User [Lohins](https://github.com/Lohins) reported the issue to CefSharp in this issue https://github.com/cefsharp/CefSharp/issues/3632 (closed)
This issue leaded to this conversation: https://github.com/cefsharp/CefSharp/discussions/3633 

In that conversation, this method is given to report the issue: https://github.com/cefsharp/CefSharp/blob/master/.github/ISSUE_TEMPLATE/bug_report.md#bug-report

To fill the report as asked by the link above is needed many information that Sentry users don't have, such as the version, architecture, code to reproduce the issue, etc., as we are not using the CefSharp directly. 

Last update: CefSharp asks for bug report by Sentry Devs, as it seems to be a bug on their side: 
https://github.com/cefsharp/CefSharp/discussions/3633#discussioncomment-3051313

**Comment by souredoutlook** [comment, marked as answer]
Hey @Alvaro-c 

Thanks for writing in and thanks for your patience while we work through this discussions backlog.

The forum post you linked to includes some discussion about updates to the getsentry/sentry-javascript package to enable the dedupe integration by default. This prevents these errors from flooding in due to bad javascript interactivity (the post alludes to the root problem stemming from Outlook Safelink Crawlers).

You can also see[ this comment](https://github.com/getsentry/sentry-javascript/issues/3440#issuecomment-878153555) regarding when we introduced the changes to integration defaults to address these bugs effect on our customers.

Can you let me know what version of the JS SDK you're using so I can follow up with the best recommendation for filtering (if you can't upgrade).

I can also see you'd like a Sentry Engineer to follow up in the[ cefsharp issue](https://github.com/cefsharp/CefSharp/discussions/3633#discussioncomment-3051313) you shared, I'll bring it to the team and see what they can do.

**Reply by lobsterkatie** [reply]
Hi, @Alvaro-c - Sentry SDK dev here.

I added a note in the original issue about filtering, etc.: https://github.com/getsentry/sentry-javascript/issues/3440

I'm also going to write back to the cefsharp guy, but TL;DR, I don't believe there's much for us to do, as we're just reporting the error that they're throwing.

**Reply by Alvaro-c** [reply]
Hi, [souredoutlook](https://github.com/souredoutlook) and [lobsterkatie](https://github.com/lobsterkatie), 
Thank you so much for your answer. 

Let me go through the different comments that all threads have and give you a more detailed answer. But as a quick update, we already have active the dedupe integration. Also, the SDK we are using is the version 7.9.0

EDIT: I continue the conversation in the issue https://github.com/getsentry/sentry-javascript/issues/3440 as it is where there is more information gathered, and also in the [cefsharp](https://github.com/cefsharp/CefSharp/discussions/3633#discussioncomment-3051313) issue to answer their questions
